src/gt4py/next/otf/compilation/build_systems/cmake_lists.py

```diff
@@ -101,8 +110,10 @@ def visit_LinkDependency(self, dep: LinkDependency):
                 lib_name = "pybind11::module"
             case "nanobind":
                 lib_name = "nanobind-static"
-            case "gridtools":
+            case "gridtools_cpu":
                 lib_name = "GridTools::fn_naive"
+            case "gridtools_gpu":
+                lib_name = "GridTools::fn_gpu"
```

Comment:
To be honest I'd reevaluate if we need all this infrastructure in the first place. The idea was the we support languages other than C++ and that we support build systems other than CMake, but we still only have GridTools C++ built with CMake, and we don't have plans to change that, neither do I see any reason to make such plans. Furthermore, in it's current form, the infrastructure does not truly support other build systems or languages in a fully generic way due to these small things like the names of the libraries hardcoded here. I think that it'd be better to massively simplify this system to handle the one and only case that actually matter to us and give up on the genericity we are not using, then we could invest the time we'd otherwise spend on coming up with generic and scalable methods into something more important.

Reply:
We decided to leave it for now and justified the decision in ADR-0016.